runtime/rasdump/dmpagent.c

```diff
@@ -2755,9 +2758,44 @@ runDumpAgent(struct J9JavaVM *vm, J9RASdumpAgent * agent, J9RASdumpContext * con
 		/* If the dump is a heap dump and exclusive access hasn't been obtained, refuse to do the dump */
 		/* This might be encapsulated more neatly if the triggering code were moved down into the dump functions themselves */
 		if (gotExclusive || (agent->dumpFn != doHeapDump)) {
+#if defined(J9ZTPF)
+			struct cujvm_dmpagent_prehook_input preHookInputParms;
+			struct cujvm_dmpagent_prehook_output preHookOutputParms;
+			struct cujvm_dmpagent_posthook_input postHookInputParms;
+			struct cujvm_dmpagent_posthook_output postHookOutputParms;
+
+			memset(&preHookInputParms, '\0', sizeof(preHookInputParms));
+			memset(&preHookOutputParms, '\0', sizeof(preHookOutputParms));
+			memset(&postHookInputParms, '\0', sizeof(postHookInputParms));
+			memset(&postHookOutputParms, '\0', sizeof(postHookOutputParms));
+#endif /* defined(J9ZTPF) */
+
 			agent->prepState = *state;
 			TRIGGER_J9HOOK_VM_DUMP_START(vm->hookInterface, vm->internalVMFunctions->currentVMThread(vm), label, detail);
+
+#if defined(J9ZTPF)
+			preHookInputParms.agentInfo = (jvmDumpAgentInfo *)agent;
+			preHookInputParms.dumpPath = label;
+			preHookInputParms.context = (jvmDumpContext *)context;
+			preHookInputParms.vm_args = (jvmInitArgs *)vm->vmArgsArray->actualVMArgs;
+			cjvm_dumpagent_preHook(&preHookInputParms, &preHookOutputParms);
+			if (NULL != preHookOutputParms.dumpPath)  {
+				if (localLabel != label) {
+					free(label);
+				}
+				label = preHookOutputParms.dumpPath;
+			}
+#endif /* defined(J9ZTPF) */
```

Comment:
Perhaps all of this addition code should go before line 2773, in particular so the dump start hook receives the updated label. For symmetry, the postHook call should go after the dump-end hook. Thoughts?

Reply:
I believe I see what you were getting at now.  I think it makes sense, given that the existing hook doesn't modify the label and z/TPF could modify it.

Update made and pushed.